packages/babel-plugin-transform-destructuring/src/index.js

```diff
@@ -120,31 +119,26 @@ export default function(api, options) {
       declar._blockHoist = this.blockHoist;
       this.nodes.push(declar);
 
-      //
-
       const tempConditional = t.conditionalExpression(
         t.binaryExpression(
           "===",
           t.identifier(tempValueRef),
           this.scope.buildUndefinedNode(),
         ),
-        pattern.right,
+        right,
         t.identifier(tempValueRef),
       );
 
-      const left = pattern.left;
       if (t.isPattern(left)) {
-        const tempValueDefault = t.expressionStatement(
-          t.assignmentExpression(
-            "=",
-            t.identifier(tempValueRef),
-            tempConditional,
-          ),
-        );
+        const tempValueRef2 = this.scope.generateUid(tempValueRef);
+
+        const tempValueDefault = t.variableDeclaration("var", [
+          t.variableDeclarator(t.identifier(tempValueRef2), tempConditional),
+        ]);
```

Comment:
Can you use `Scope#push`, since it also adds the new binding to the scope?

Reply:
I wound up cleaning up the code a bit, which nuked your comment, sorry!

Using `Scope#push` seems like it'd be a larger change to `DestructuringTransformer` than I was hoping for this PR, as it's currently collecting the nodes that get added for [processing later](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-destructuring/src/index.js#L553)... maybe a followup to revisit/refactor?